src/ray/gcs/client_test.cc

```diff
@@ -200,15 +228,19 @@ void TestTableLookupFailure(const JobID &job_id,
   test->Start();
 }
 
-TEST_F(TestGcsWithAe, TestTableLookupFailure) {
-  test = this;
-  TestTableLookupFailure(job_id_, client_);
-}
+#define TEST_TABLE_LOOKUP_FAILURE(FIXTURE)    \
```

Comment:
Maybe pass in the test name to the macro as well? Something like: `#define TEST_TABLE(FIXTURE, TEST)`?

Reply:
Are you proposing we do something like the following?
```
#define TEST_TABLE(FIXTURE, TEST) \
  TEST_F(FIXTURE, TEST) {         \
    test = this;                  \
    TEST(job_id_, client_);       \
  }

TEST_MACRO(TestGcsWithAe, TestTableLookupFailure);
TEST_MACRO(TestGcsWithAsio, TestTableLookupFailure);
#if RAY_USE_NEW_GCS
TEST_MACRO(TestGcsWithChainAe, TestTableLookupFailure);
TEST_MACRO(TestGcsWithChainAsio, TestTableLookupFailure);
#endif
<and other tests>
```

If so, should I refactor the existing TEST_F in this file that is not affected by this PR?  For example, the current state of the PR left 
```
TEST_F(TestGcsWithAe, TestLogLookup) {
  test = this;
  TestLogLookup(job_id_, client_);
}

TEST_F(TestGcsWithAsio, TestLogLookup) {
  test = this;
  TestLogLookup(job_id_, client_);
}
```
untouched.

Reply:
Yes, I think we should just refactor the other `TEST_F`s as well.

Reply:
Done